City-of-Helsinki/kukkuu-admin
# KK-762 && KK-771 | Support for external ticket systems

## Description

Implemented changes needed to support events with an external ticket system (only option Ticketmaster ATM).

Changes:
* When creating and editing an unpublished event, one can choose whether its ticket system is the internal one or an external one (Ticketmaster). The ticket system cannot be changed after the event has been published.
* Event / event group list view's total capacity field shows value "unknown" if any of the event groups events is a Ticketmaster event.
* How Ticketmaster events differentiate from internal ticket system events:
    * there is no need to set capacity so the capacity field is hidden in the forms and the detail view.
    * occurrence create and edit views have an input for setting Ticket system URL. Also, the capacity input is hidden.
    * occurrence list and detail views don't include capacity, participants or number of subscriptions, and have a field for the Ticketmaster URL instead.

Known glitch: changing ticket system in event create view alters the form's width a bit.

Also, it turned out that this whole feature isn't going to be needed at least in the near future, so the whole thing is put under a feature flag.

## Context

[KK-762](https://helsinkisolutionoffice.atlassian.net/browse/KK-762)

## How Has This Been Tested?

There are no automated tests for this yet, those should be implemented if/when this actually goes into use.

Tested manually by intensively clicking around.

## Manual Testing Instructions for Reviewers

Create an event (and some occurrences) with Ticketmaster as the ticket system, and verify by viewing and editing stuff that the listed changes work as expected. Try also changing an existing event's ticket system type between internal and Ticketmaster, that was causing issues during development.
